relayer/chains/cosmos/query.go

```diff
@@ -738,7 +738,7 @@ func (cc *CosmosProvider) QueryRecvPacket(
 	}
 	for _, msg := range ibcMsgs {
 		if pi, ok := msg.info.(*packetInfo); ok {
-			if pi.Sequence == sequence {
+			if pi.DestChannel == dstChanID && pi.Sequence == sequence {
```

Comment:
Is this actually necessary? Shouldn't the `ibcMsgs` already only contain the packets for this particular channel since we are setting the destination channel in the call to `writeAcknowledgementQuery` on L734?

Reply:
Yeah, At first I think it is not necessary as well , But I met some problems in mainnet.
case:
A,B,C three chains
t1: A(channel-20)<=>B(channel-14) packet  2 is not relayed  and the connection between A and C is gone forever
t2: B(channel-30)<=> C(channel-14) packet 2 is not relayed too 
t3:,when you try to relay packet 2  between B and C ,you will execute QueryRecvPacket which returns two packets. unfortunately, the two packets contain the t1 packet which is always invalid for chain C

Reply:
Besides, As 'transfer' is not ORDERED ,I do think maybe we can shuffle the sequence to skip the bad sequence and when you search the packet one by one ,maybe it is not a good idea to return asap when one of them is failed

Reply:
Ahh interesting, I guess taking another look at it we have to explicitly check the sequence number after the query also, even though we are supposedly only querying for one sequence so it makes sense that we need to check the channel id too. Good catch!!